### PR TITLE
Issue 2015

### PR DESCRIPTION
**Pull request by b00f**
* [ ] Updated all relevant documentation in docs
* [X ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

Here is a fix for issue #2015 
Note: for testing this issue I used my local testnet with 4 nodes. you can find out a copy of it here: https://github.com/b00f/tm-testnet

I manually changed the genesis file in new node and removed all validators. Then I updated the BaseApplication (InitChain phase) for updating the state. It worked properly.

### CODE REVIEW

File: types/genesis_test.go [hunk 1]
```diff
@@ -14,14 +14,10 @@ import (
 func TestGenesisBad(t *testing.T) {
 	// test some bad ones from raw json
 	testCases := [][]byte{
-		[]byte{},                                           // empty
-		[]byte{1, 1, 1, 1, 1},                              // junk
-		[]byte(`{}`),                                       // empty
-		[]byte(`{"chain_id":"mychain"}`),                   // missing validators
-		[]byte(`{"chain_id":"mychain","validators":[]}`),   // missing validators
```
Comment (melekes) [inline types/genesis_test.go]: can we just move these test cases into TestGenesisGood? in order to test that they don't result in error

Comment (melekes) [inline types/genesis_test.go]: can we not remove these, but move them to TestGenesisGood instead?

File: abci/types/application.go [hunk 2]
```diff
@@ -63,6 +63,26 @@ func (BaseApplication) Query(req RequestQuery) ResponseQuery {
 
 func (BaseApplication) InitChain(req RequestInitChain) ResponseInitChain {
 	return ResponseInitChain{}
+	/*
+		This part is for code review. How I update validator list?
```
Comment (melekes) [inline abci/types/application.go]: Can you clarify your question? Do you want to update a running chain (https://tendermint.com/docs/app-dev/abci-spec.html#endblock)?

Comment (b00f) [reply abci/types/application.go]: I am so sorry. I didn't mean to ask. It's not a question. I wanted to show that how I tested it... ignore the question mark! ;)

File: abci/types/application.go [hunk 3]
```diff
@@ -63,6 +63,28 @@ func (BaseApplication) Query(req RequestQuery) ResponseQuery {
 
 func (BaseApplication) InitChain(req RequestInitChain) ResponseInitChain {
 	return ResponseInitChain{}
+	/*
```
Comment (melekes) [inline abci/types/application.go]: can you remove this comment?

File: types/genesis_test.go (hunk 1 above)
Comment (b00f) [reply types/genesis_test.go]: They supposed to be bad genesis json? shall I move them to `GoodGenesisTest`?
I added new test with wrong public key.

Comment (b00f) [reply types/genesis_test.go]: Done. Please check the latest commit.

Comment (melekes) [reply types/genesis_test.go]: Oh god. I think you misunderstood me. I wanted you to only move "missing validators" lines, not the whole `TestGenesisBad` test. The reason for moving is that "missing validators" are now correct use-case, so imho should be inside `TestGenesisGood`

Comment (b00f) [reply types/genesis_test.go]: Please check it now

**Pull request merged**